compiler/noirc_evaluator/src/ssa/ir/instruction/call.rs

```diff
@@ -730,7 +730,7 @@ mod tests {
                 return v2
             }
             "#;
-        let ssa = Ssa::from_str(src).unwrap();
+        let ssa = Ssa::from_str_simplifying(src).unwrap();
```

Comment:
Why do we simplify in these situations?

Reply:
I think it's just this particular test that is actually about the correctness of the simplification.

Reply:
I see there is another test below where this is less obvious (`simplify_derive_generators_has_correct_type` vs `constraint_decomposition`). 

Just wondering if it would make sense to add a method to the `Ssa` which re-inserts all instructions with simplification? As I understand some mechanism actually assume that this will happen sooner or later, like what we discussed in https://github.com/noir-lang/noir/issues/6842 Having an explicit way to trigger this would perhaps not be out of place, and then a test that needs it can parse the SSA without simplification and then call this method. Then the "simple" act of parsing a piece of SSA doesn't need to be burdened with the knowledge of further effects we want to see happen.

Reply:
I don't know if this would be what the "old" normalize pass would do. So maybe this could be done by having the normalize pass receive a boolean flag. We could try it in a follow-up PR.

Reply:
Yeah, it's essentially bringing back that old behaviour.

I'd argue that if we're only using this in a couple of tests then having the "special behaviour" only in test code is a good thing (and so we shouldn't make that follow-up). We could quite possibly refactor these tests to not rely on this anyway.